### ISSUE TITLE
container does not clear

### ISSUE THREAD

**Issue by mukral** [zauberzeug/nicegui#191]
Hi,
I need to clear a container it during running the program and not - like in the example - by clicking a button. I've been trying for several hours (days) now but I could not get it work (because of my limited programming skills...).

Can someone please give me an expample how to make it work?
```python3
from nicegui import ui
container1 = ui.row()
container2 = ui.row()
with container1:
   sphereRadius = ui.number(label='radius', value=2)
def drawScene(sphereRadius):
    clearContainer
    with container2:
        with ui.scene(width=640, height=400) as scene:
            scene.sphere(sphereRadius)
def clearContainer():
    container2.clear()
ui.button("go", on_click=lambda: drawScene(sphereRadius.value))
ui.button("clear", on_click=clearContainer)
ui.run()
```

Thanks a lot
Markus

**Comment by falkoschindler** [comment]
On a first glance I notice missing brackets when calling `clearContainer` within `drawScene`. But still, the container is not updating.

I managed to get it working with a short delay after clearing:
```python
async def drawScene():
    container2.clear()
    await asyncio.sleep(0.01)
    with container2:
        with ui.scene(width=640, height=400) as scene:
            scene.sphere(sphereRadius.value)

ui.button("go", on_click=drawScene)
```

The issue seems to be gone in the upcoming NiceGUI version 1.0. In the meantime, you can hopefully work around the problem with the `sleep()` hack.

**Comment by falkoschindler** [comment]
This issue should be fixed in NiceGUI 1.0.

**Comment by mukral** [comment]
Hi, thank you very much for your example - it works fine and I've tried to adapt it to another test and could not get it running.
Similar problem and sorry I haven't found a way how to integrate the async/await functions - I've really tried hard. Sometimes I got a "coroutine 'drawScene' was never awaited", sometimes "cannot run in a eventhandler", sometimes nothig happened at all - I totally understand that the machine has conflicts - does anybody knows how to avoid it?
```
import cadquery as cq
from nicegui import ui
import asyncio
container1 = ui.row()
container2 = ui.row()
with container1:
   sphereRadius = ui.number(label='radius', value=2)
   sphereColor = ui.number(label='color', value=1)
def newSphereSTL():
    sphere1 = cq.Workplane().sphere(sphereRadius.value)
    cq.exporters.export(sphere1, 'C:/.../static/testSphereSTL.stl')
    newScene()
    #asyncio.run(drawScene())
#async def drawScene():
def drawScene():
    #container2.clear()
    ui.add_static_files('/static', 'C:/.../static')
    #await asyncio.sleep(0.01)
    with container2:
        with ui.scene(width=640, height=400) as scene:
            testSphereSTL = 'static/testSphereSTL.stl'
            scene.stl(testSphereSTL).scale(1).material('#00ff00')
#            scene.sphere(sphereRadius.value)
    print('container2 updated')

ui.button("go", on_click=newSphereSTL)
ui.run()
```
Thank you, Markus
Hopefully 1.0 is coming soon...

**Comment by falkoschindler** [comment]
As far as I can tell without running the code (which is tricky without the STL files):

- Use `await drawScene()` to call an async function (instead of using `asyncio.run`).
- Define `newSphereSTL` as `async def newSphereSTL():` to support the `await` expression.
- Adding static files again with each button click seems odd. Do it once in global scope.

By the way: We released NiceGUI 1.0 just yesterday. So feel free to try it out.
